src/test/java/org/opensearch/knn/integ/AutoIndexCleanUpTestCase.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.integ;
+
+import lombok.extern.log4j.Log4j2;
+import org.junit.After;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.knn.KNNRestTestCase;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * This class provides the convenience of automatically deleting indices created during tests once they are completed
+ */
+@Log4j2
+public class AutoIndexCleanUpTestCase extends KNNRestTestCase {
```

Comment:
Why make this a test case? Can you create a utility class and plug it in the required test. Deep hierarchies are not good for code reuse, this is adding on to already existing hierarchies. 

Cleanup can be different for each test so whatever tests require it can call the utility by passing in indices